blocks/image-compare/src/edit.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * WordPress dependencies
+ */
+import { InspectorControls, RichText } from '@wordpress/block-editor';
+import { SelectControl, TextControl, Placeholder } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import UploadPlaceholder from './upload-placeholder';
+
+const edit = ( { attributes, className, isSelected, setAttributes } ) => {
+	const { imageBefore, imageAfter, caption, orientation } = attributes;
+
+	// if both are defined, add juxtaspose class
+	// the juxtapose library uses class when page is scanned
+	// to find the images and apply the side-by-side magic
+	const cls = ( imageBefore && imageAfter ) ? 'juxtapose' : 'controls';
+
+	return (
+		<>
+			<InspectorControls key="controls">
+				<h4> Orientation </h4>
+				<SelectControl
+					value={ orientation }
+					options={ [
+						{ label: 'Side by Side', value: 'horizontal' },
+						{ label: 'Above and Below', value: 'vertical' },
+					] }
+					onChange={ ( val ) => {
+						setAttributes( { orientation: val } );
+						// need slight delay so markup can be updated before
+						// scan page gets triggered
+						setTimeout( function() {
+							juxtapose.scanPage();
+						}, 100 );
+					} }
+				/>
+			</InspectorControls>
+			<div className={ cls } data-mode={ orientation }>
+
+				<Placeholder>
+					{ imageBefore ? (
+						<img src={ imageBefore } />
```

Comment:
We need an alt attribute. If it can't be dynamic, "First image" or something in that vein is fine.